data_models/aws_cloudtrail_data_model.py

```diff
@@ -6,8 +7,26 @@ def get_event_type(event):
     # currently, only tracking a few event types
     if event.get('eventName') == 'ConsoleLogin' and deep_get(
             event, 'userIdentity', 'type') == 'IAMUser':
-        if event.get('responseElements', {}).get('ConsoleLogin') == 'Failure':
+        if deep_get(event, 'responseElements', 'ConsoleLogin') == 'Failure':
             return event_type.FAILED_LOGIN
-        if event.get('responseElements', {}).get('ConsoleLogin') == 'Success':
+        if deep_get(event, 'responseElements', 'ConsoleLogin') == 'Success':
             return event_type.SUCCESSFUL_LOGIN
     return None
+
+
+def load_ip_address(event):
+    '''
+    CloudTrail occassionally sets non-IPs in the sourceIPAddress field.
+    This method ensures that either an IPv4 or IPv6 address is always returned.
+    '''
+    source_ip = event.get('sourceIPAddress')
+    if not source_ip:
+        return None
+    try:
+        ip = ipaddress.IPv4Address(source_ip)
+    except ipaddress.AddressValueError:
+        try:
+            ip = ipaddress.IPv6Address(source_ip)
+        except ipaddress.AddressValueError:
+            return None
+    return ip
```

Comment:
This now returns a `ipaddress.IPv4Address` or `ipaddress.IPv6Address` object instead of a string. That's going to break any detections that are currently relying on `event.udm('source_ip')` returning a string. It also means this data model does not return the same type as the rest of the data models.

Now as it happens python's loose typing means that all existing rules still work (I did a quick check locally), but the inconsistency is exactly against what the data models are for so I don't think we should move forward like that.

So the options as I see them are:
1. Change this function to return `source_ip`, a string, and then everything stays the same
2. Change all the other data models to return an `IPv4Address` or `IPv6Address` object when `source_ip` is queried
3. Add a new UDM field, `source_ip_extracted` that returns an `IPv4Address` object, and leave `source_ip` to be a string

The first is the easiest, but I like the idea of the second. If you want to go with the first for now, we can put up an issue to change to the second next sprint.